ggml.c

```diff
@@ -1586,7 +1586,7 @@ inline static void ggml_vec_argmax_f32(const int n, int * s, const float * x) {
 // data types
 //
 
-static const char * GGML_OP_NAME[GGML_OP_COUNT] = {
+static const char * GGML_OP_NAME[GGML_OP_COUNT + 2] = {
```

Comment:
I don't see a warning about this when building with CMake and gcc 11.3. This seems like the wrong fix - if we are actually missing operators then OP_COUNT should be adjusted and the actual names should be added to the list. What is the warning that you are seeing?

Reply:
Ah, I see it with gcc 13:
```
In function ‘ggml_op_name’,
    inlined from ‘ggml_get_n_tasks’ at /home/jared/src/forks/llama.cpp/ggml.c:15698:17:
/home/jared/src/forks/llama.cpp/ggml.c:2019:24: warning: array subscript 69 is above array bounds of ‘const char *[68]’ [-Warray-bounds=]
 2019 |     return GGML_OP_NAME[op];
      |            ~~~~~~~~~~~~^~~~
/home/jared/src/forks/llama.cpp/ggml.c: In function ‘ggml_get_n_tasks’:
/home/jared/src/forks/llama.cpp/ggml.c:1589:21: note: while referencing ‘GGML_OP_NAME’
 1589 | static const char * GGML_OP_NAME[GGML_OP_COUNT] = {
      |                     ^~~~~~~~~~~~
```

The problem is that we are calling ggml_op_name in ggml_get_n_tasks when the op is not recognized:

https://github.com/ggerganov/llama.cpp/blob/e937066420b79a757bf80e9836eb12b88420a218/ggml.c#L15696-L15698

We should either not attempt to print the name, or only attempt to print it if 'op' is less than GGML_OP_COUNT.

Reply:
>  only attempt to print it if 'op' is less than GGML_OP_COUNT.

Yes.

Reply:
The current code looks correct to me. It's kind of hard to make compiler happy in this case :-(

I'll take it out for now